activerecord/lib/active_record/destroy_association_async_job.rb

```diff
@@ -4,23 +4,32 @@ module ActiveRecord
   class DestroyAssociationAsyncError < StandardError
   end
 
-  # Job to destroy the records associated with a destroyed record in background.
   class DestroyAssociationAsyncJob < ActiveJob::Base
     queue_as { ActiveRecord.queues[:destroy] }
 
     discard_on ActiveJob::DeserializationError
 
     def perform(
-      owner_model_name: nil, owner_id: nil,
-      association_class: nil, association_ids: nil, association_primary_key_column: nil,
-      ensuring_owner_was_method: nil
-    )
+          owner_model_name: nil,
+          owner_id: nil,
+          association_class: nil,
+          association_ids: nil,
+          association_primary_key_column: nil,
+          ensuring_owner_was_method: nil,
+          ensuring_owner_destroyed: true
```

Comment:
Adding a new parameter to this job is tricky and should be avoided if we can. If we add a new option what can happen is that during deploy a working with the old version of the job will load the job and will fail because the number of parameters is bigger than the expected in the job. Do we need this new parameter?

Reply:
If that the case even if there's a default value for the new parameter?

Reply:
@rafaelfranca The parameter is needed for the case when the association is destroyed, but the parent is not. I'm not sure if we can recognize it without the param.

@ghiculescu yes, because running, old code might expect a **smaller** number of params.

Reply:
@rafaelfranca ping --^

Reply:
@rafaelfranca ping --^

Reply:
In this case unfortunately we need to think in a way to make this work with older versions of this job. Maybe we should change the job signature to accept `**options` in 7.0 and in 7.1 we can merge this PR.

Reply:
Can't we add the `**options` now instead of that new param? Then, we could release the fix for 6.x too.

Reply:
There is no guarantee people will update to 6.1.X, where X is the commit that will accept `**options` before going to 7.0. If they don't do that they will get exceptions in production when deploying Rails 7.0, so I prefer to play safe and not introduce a breaking changing that we can automatically recover from in Rails 7.0.

Reply:
What should I do with this PR? Wait for 7.0, rebase and ping for review or close?

Reply:
Let's open a new PR with the `**options` introduction that we can try to backport to 6.1. Then I'll merge this to be released in 7.0.0.beta1. If we get complain we revert.

Reply:
👍, I will prepare the PR tomorrow.

Reply:
https://github.com/rails/rails/pull/43736

I will rebase this PR when #43736 is approved.